### PR TITLE
Project object assignment

### PR DESCRIPTION
**Pull request by hypebright**
I decided to create one app.R file instead of separate ui, server and global files. At the top level of the app.R file, we initiate the proj_obj with:

```r
proj_obj <- get_proj()
```

Then in the ui part we pass along this proj_obj to the modules. This means that the modules all got new arguments.

On the server side we handle it a bit differently. We get the initiated project, and we put it in the petit `r`:

```r
  # Top-level reactive values  
  r <- reactiveValues(active_tab = "",
                      model_updated = 0,
                      proj_obj = get_proj())
```

throughout the server modules we’ll be referencing `r$proj_obj`, or we will be updating it with `r$proj_obj ← get_proj()`

The environment stays clean. This is because the top-level of an app.R file is “global” to the Shiny session (and accessible from ui and server), but it doesn’t pollute the user’s global environment. This also means we can get rid of all the assign statements.

Because we’re running a Shiny app in another location, we need to some magic with working directories, so I introduced a `this_wd` variable that sets the working directory to the project folder. We store this in `r` too and  his is passed along in the `get_proj()` function each time.

WIP.

### CODE REVIEW

File: R/get_proj.r [hunk 1]
```diff
@@ -27,13 +27,6 @@ get_proj <- function(projloc=".",geteval=TRUE){
   datf  <- list.files(paste0(projloc,"/shinyMixR/app/data"))
   grepd <- " |^[[:digit:]]|\\!|\\#|\\$|\\%|\\&|\\'|\\(|\\)|\\-|\\;|\\=|\\@|\\[|\\]|\\^\\`\\{\\|\\}"
   if(any(grepl(grepd,datf))) warning("Data files with special characters found, take into acount that models that use these can crash")
-  # not relevant to read all data for running nlmixr in separate session (should be loaded in this session!)
-  lapply(list.files(paste0(projloc,"/shinyMixR/app/data"),full.names = TRUE),function(x){
-    if(!grepl(grepd,x) & !exists(sub("\\.rds$|\\.csv$","",basename(x),ignore.case = TRUE),envir=.GlobalEnv)){
-      if(grepl("\\.rds$",x,ignore.case = TRUE)) assign(sub("\\.rds$","",basename(x),ignore.case = TRUE),readRDS(x),pos = .GlobalEnv)
-      if(grepl("\\.csv$",x,ignore.case = TRUE)) assign(sub("\\.csv$","",basename(x),ignore.case = TRUE),read.csv(x),pos = .GlobalEnv)
-    }
-  })
```
Comment (hypebright) [inline R/get_proj.r]: @RichardHooijmaijers I don't think we need this. I can't find any locations the app where we are pulling the data from the global environment. When we do need this data, we check if it's there, but if it's not, we pull it from the RDS files. So I think we can leave this out. At the same we get rid of the global assignments- they needed to go anyway.

File: tests/testthat/test-shinymixr-03-create-newmodel.R
Comment (RichardHooijmaijers) [inline tests/testthat/test-shinymixr-03-create-newmodel.R:1]: Test is rewritten to fit new structure

File: NAMESPACE
Comment (RichardHooijmaijers) [inline NAMESPACE:1]: I had some issues with this one... I wanted to add nlmixr2 as import. I was hoping that when running `document()` it was automatically added to NAMESPACE. This was not the case, so I added it manually, but was removed when running `document` again. I guess I am doing something wrong here but like to briefly discuss this issue (noticed you added manual imports as well, right?)

**Review by RichardHooijmaijers — COMMENTED** [review]
I hope everything is clear.. Added some comments and pushed quite a lot of changes in this PR. Just have a look and see what you think. For me most things are OK now (especially the main reason for this PR; do not use global assignment). Will add some comments to notion

File: NAMESPACE
Comment (hypebright) [reply NAMESPACE:1]: @RichardHooijmaijers the NAMESPACE file should never be edited manually! If you want it to end up there, you need to include it in the roxygen comments of the function that uses the function (at least once in one of your functions), and then run `devtools::document()`. Alternatively, if you have things that need to be imported "globally", you can use the s`hinyMixR-package.R` file for that. If you add new `importFrom` statements there, and run `devtools::document()`, they also ed up in NAMESPACE.

File: R/exploreplot.r
[the remark this answers is not in the document]
Comment (hypebright) [reply R/exploreplot.r:1]: Oops, typo 😅

File: R/run_nmx.r
[the remark this answers is not in the document]
Comment (hypebright) [reply R/run_nmx.r:1]: @RichardHooijmaijers as soon as there is `assign` it's a no-go 😬 .

**Comment by RichardHooijmaijers** [comment]
looks good, let's finish it up!

**Pull request merged**